contracts/governance/Vesting/VestingRegistryLogic.sol

```diff
@@ -247,27 +247,26 @@ contract VestingRegistryLogic is VestingRegistryStorage {
 		uint256 teamVestingType = 0;
 		uint256 uid;
 		uint256 length = vestingRegistries.length;
-		for (uint256 i = 0; i < length; i++) {
-			address vestingAddress = vestingRegistries[i].getVesting(_tokenOwner);
-			if (vestingAddress != address(0)) {
-				VestingLogic vesting = VestingLogic(vestingAddress);
-				uid = uint256(
-					keccak256(abi.encodePacked(_tokenOwner, vestingType, vesting.cliff(), vesting.duration(), _vestingCreationType))
-				);
-				vestings[uid] = Vesting(vestingType, _vestingCreationType, vestingAddress);
-				vestingsOf[_tokenOwner].push(uid);
-				isVesting[vestingAddress] = true;
-			}
-			address teamVestingAddress = vestingRegistries[i].getTeamVesting(_tokenOwner);
-			if (teamVestingAddress != address(0)) {
-				VestingLogic vesting = VestingLogic(teamVestingAddress);
-				uid = uint256(
-					keccak256(abi.encodePacked(_tokenOwner, teamVestingType, vesting.cliff(), vesting.duration(), _vestingCreationType))
-				);
-				vestings[uid] = Vesting(teamVestingType, _vestingCreationType, teamVestingAddress);
-				vestingsOf[_tokenOwner].push(uid);
-				isVesting[teamVestingAddress] = true;
-			}
+		uint256 i = _vestingCreationType - 1;
```

Comment:
You didn't need the loop here as you could add vestings for the same `_tokenOwner` multiple times, did you?

Reply:
This function is only used for migrating previous vesting registries to the new upgradable vesting registry. Hence, instead of looping through all previous registries as highlighted here, we passed just that one registry at a time(using _vestingCreationType field). For example, a token owner only has previous vesting is Registry 1 and Registry 3, so we can skip processing for Registry 2 and save gas.